src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1247,44 +1251,72 @@ public function seek( $bookmark_name ) {
 		$bookmark_starts_at   = $this->bookmarks[ $actual_bookmark_name ]->start;
 		$direction            = $bookmark_starts_at > $processor_started_at ? 'forward' : 'backward';
 
-		switch ( $direction ) {
-			case 'forward':
-				// When moving forwards, reparse the document until reaching the same location as the original bookmark.
-				while ( $this->step() ) {
-					if ( $bookmark_starts_at === $this->bookmarks[ $this->state->current_token->bookmark_name ]->start ) {
-						return true;
-					}
+		/*
+		 * If seeking backwards, it's possible that the sought-after bookmark exists within an element
+		 * which has fully closed at the point of the current cursor. This means that the breadcrumbs
+		 * will be wrong after seeking because the chain of open elements leading to that bookmark have
+		 * been discarded.
```

Comment:
I find "fully closed" confusing. It implies some elements are in different degrees of closed which I don't think is the case. Do we mean "not on the stack of open elements"?

Some ideas here:

```suggestion
		 * If seeking backwards, it's possible that the sought-after bookmark is within an element
		 * that is not on the stack of open elements. This means that it's insufficient to pop elements off the 
		 * stack of open elements until the element at the top of the stack of open elements is before the sought-after bookmark and then jump to the bookmark because 
		 * because the chain of open elements leading to that bookmark will not include the element was not initially on the stack of open elements.
```

Reply:
Reworded, thank you.